services/brig/src/Brig/Run.hs

```diff
@@ -103,3 +124,51 @@ lookupRequestIdMiddleware :: (RequestId -> Wai.Application) -> Wai.Application
 lookupRequestIdMiddleware mkapp req cont = do
   let reqid = maybe def RequestId $ lookupRequestId req
   mkapp reqid req cont
+
+cleanExpiredPendingInvitations :: AppIO ()
+cleanExpiredPendingInvitations = do
+  safeForever "cleanExpiredPendingInvitations" $ do
+    today <- utctDay <$> (liftIO =<< view currentTime)
+    for_ [0 .. 9] $ \i ->
+      cleanUpDay (addDays (- i) today)
+    let d :: Int = 24 * 60 * 60
+    randomSecs <- liftIO (round <$> randomRIO @Double (0.5 * fromIntegral d, fromIntegral d))
+    threadDelay (randomSecs * 1_000_000)
+  where
+    cleanUpDay :: Day -> AppIO ()
+    cleanUpDay day = do
+      expiredEntries <- forTrackedExpirations day $ \exps ->
+        catMaybes
+          <$> ( for exps $ \(PendingActivationExpiration uid expiresAt tid) -> do
+                  isExpired <- (expiresAt <=) <$> (liftIO =<< view currentTime)
+                  if isExpired
+                    then do
+                      isPendingInvitation <- (Just PendingInvitation ==) <$> lookupStatus uid
+                      invExpired <- isNothing <$> Data.lookupInvitation tid (coerce uid)
```

Comment:
No need to do this:

```suggestion
```

This is true by construction.

Reply:
I think `isPendingInvitation <- (Just PendingInvitation ==) <$> lookupStatus uid` can also removed by the same argument

Reply:
I removed both conditions

Reply:
You're right, didn't think about that.  Yeay, pairing!  :)

Reply:
Checking for invitations with `invExpired <- isNothing <$> Data.lookupInvitation tid (coerce uid)` is required, because I chose not to call `removeTrackedExpiration` when a user accepts their invitation (since it was necessary because of this check).
Even if we decide to call `removeTrackedExpiration` when the user accepts their invitation, we still might want to keep this check as a safety measure.

Reply:
Err, checking for missing inivitation really doesn't make any sense, since invitations also get removed when a user accepts them. Checking for `PendingInvitation` however provides an additional guar. Thanks @jschaul